ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v1
+
+import java.time.{Duration, Instant}
+
+trait TimeModel {
+
+  //TODO docs
+  def minTransactionLatency: Duration
+
+  def maxClockSkew: Duration
+
+  def minTtl: Duration
+
+  def maxTtl: Duration
+
+  /**
+    * Validates that the given ledger effective time is within an acceptable time window of the current system time.
+    *
+    * @param currentTime              the current time
+    * @param givenLedgerEffectiveTime The ledger effective time to validate.
+    * @param givenMaximumRecordTime   The maximum record time to validate.
+    * @return true if successful
+    */
+  def checkLet(
```

Comment:
what do you think about just having the "TimeModel" data here and have the checks separate from it? e.g. https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto#L127.

Reply:
good idea, will do that.